librocksdb-sys/rocksdb-c.cc

```diff
@@ -0,0 +1,292 @@
+#include "rocksdb-c.h"
```

Comment:
I think it's a bad idea to add this code. I suppose you should create a PR to https://github.com/facebook/rocksdb/ first. What if I'd like to link with system `librocksdb` rather than compile from the sources?

Reply:
> I think it's a bad idea to add this code. I suppose you should create a PR to https://github.com/facebook/rocksdb/ first. What if I'd like to link with system `librocksdb` rather than compile from the sources?

I have created a PR to rocksdb but get no response after a long time (see https://github.com/facebook/rocksdb/pull/9252). So I chose to implement these in rust-rocksdb. And I guess `rust-rocksdb` doesn't support link with system `librocksdb` currently.

Reply:
On macOS, if I build from master I can link with system `librocksdb` without any problem.
```
 % brew info rocksdb                                                                                                                                                                                                                          
rocksdb: stable 7.0.3 (bottled), HEAD
```

Reply:
> On macOS, if I build from master I can link with system `librocksdb` without any problem.
> 
> ```
>  % brew info rocksdb                                                                                                                                                                                                                          
> rocksdb: stable 7.0.3 (bottled), HEAD
> ```

In this case, maybe we can just compile the C API part of rocksdb and link with system `librocksdb`? But considering that `librocksdb` may use a different C++ compiler, I don't know if this can guarantee compatibility.

Reply:
> I think it's a bad idea to add this code. I suppose you should create a PR to https://github.com/facebook/rocksdb/ first. What if I'd like to link with system `librocksdb` rather than compile from the sources?

https://github.com/facebook/rocksdb/commit/b71466e9821579592c3324e6b44ad529f2a4f052

Reply:
> I think it's a bad idea to add this code. I suppose you should create a PR to https://github.com/facebook/rocksdb/ first. What if I'd like to link with system `librocksdb` rather than compile from the sources?

see 
https://github.com/facebook/rocksdb/commit/b71466e9821579592c3324e6b44ad529f2a4f052 , the code seems merged into rocksdb

Reply:
Any progress? Need to wait for the rocksdb release that contains this update?

Reply:
RocksDB 7.3.1 [Latest](https://github.com/facebook/rocksdb/releases/latest)
released

Reply:
> RocksDB 7.3.1 [Latest](https://github.com/facebook/rocksdb/releases/latest) released

Unfortunately, this release version doesn't contain the required changes, it added some bug fixes compared to 7.3.0. May have to wait 7.4.

Reply:
Any luck with 7.4.3 ?

Reply:
updated